src/main/java/net/minecraftforge/fml/common/registry/EntityEntryBuilderImpl.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.fml.common.registry;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityList;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.entity.EnumCreatureType;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraft.world.biome.Biome;
+import net.minecraftforge.fml.common.Loader;
+import net.minecraftforge.fml.common.ModContainer;
+import net.minecraftforge.fml.relauncher.ReflectionHelper;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class EntityEntryBuilderImpl<E extends Entity> implements EntityEntryBuilder<E>, EntityEntryBuilder.Id<E>, EntityEntryBuilder.Name<E>, EntityEntryBuilder.Rest<E>, EntityEntryBuilder.Tracking<E>
+{
+    private final ModContainer mod;
+    private Class<? extends E> entity;
+    private ResourceLocation id;
+    private int network;
+    private String name;
+    private int trackingRange;
+    private int trackingUpdateFrequency;
+    private boolean trackingVelocityUpdates;
+    @Nullable private Collection<Spawn> spawns;
+    @Nullable private EntityList.EntityEggInfo egg;
+
+    // Create an instance of this class using EntityEntryBuilder.create()
+    protected EntityEntryBuilderImpl()
+    {
+        this.mod = Loader.instance().activeModContainer();
+    }
+
+    @Nonnull
+    @Override
+    public Id<E> entity(@Nonnull final Class<? extends E> entity)
+    {
+        this.entity = checkNotNull(entity, "entity class");
```

Comment:
These methods should check if it's actually good to call them right now. People might ignore the return values from these methods and call methods on the "old" builder instances.

Reply:
You can't, the return type doesn't contain the old methods.